src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/ExecutionServiceCluster.scala

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkflowRecord
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.concurrent.{Future}
+import scala.util.Try
+
+/**
+  * Created by davidan on 6/14/16.
+  */
+trait ExecutionServiceCluster extends ErrorReportable {
+  override val errorReportSource = "cromwell"
+
+  // ====================
+  // facade methods
+  // ====================
+  def submitWorkflow(wdl: String, inputs: String, options: Option[String], userInfo: UserInfo): Future[ExecutionServiceStatus]
+
+  def submitWorkflows(workflowRecs: Seq[WorkflowRecord], wdl: String, inputs: Seq[String], options: Option[String], userInfo: UserInfo): Future[(ExecutionServiceId, Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]])]
+
+  def status(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceStatus]
+
+  def callLevelMetadata(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionMetadata]
+
+  def outputs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceOutputs]
+
+  def logs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceLogs]
+
+  def abort(wfe: WorkflowExecution, userInfo: UserInfo): Future[Try[ExecutionServiceStatus]]
+
+}
+
+case class WorkflowExecution(
+  id: String,
+  executionServiceId: Option[String]
+)
+
+// we need the external id in order to query cromwell, and the execution service id in order to route to the right instance
+object WorkflowExecution {
+  def apply(wf: Workflow) = wf.workflowId match {
+    case Some(id) => new WorkflowExecution(id, None)
+    case None => throw new RawlsException("can only process Workflow objects with a workflow id")
+  }
+  def apply(wr: WorkflowRecord) = (wr.externalId, wr.executionServiceKey) match {
+    case (Some(id), x) => new WorkflowExecution(id, x) // legacy workflows will have None for executionServiceKey
```

Comment:
also for @dvoet due to comment above - yes, you're reading that right. Any workflow run prior to this sharding code will have null for exec_service_key in the db, and will fall through the code, eventually using the default cromwell instance (which we define as the one we've used already). I thought this was best - that way we can differentiate between old workflows and new/sharded workflows. But maybe we don't need that - and we could just update the db to mark each of the  old workflows.